src/test/test/controller.test.ts

```diff
@@ -139,6 +139,26 @@ describe("controller", () => {
       assert.equal(getDataStub.calledOnce, false);
       assert.equal(setItemsStub.calledOnce, false);
     });
+
+    it(`10: should quickPick.show and quickPick.loadItems and quickOpen.setText method be invoked
+      if shouldQuickPickOpenFromSelection method return true and active text editor found`, async () => {
+      const [loadItemsStub, showStub, setTextStub] = setups.search10();
+      await controller.search();
+
+      assert.equal(loadItemsStub.calledOnce, true);
+      assert.equal(showStub.calledOnce, true);
+      assert.equal(setTextStub.calledOnce, true);
+    });
+
+    it(`11: should quickPick.show and quickPick.loadItems invoked, and quickOpen.setText method not be invoked
+      if shouldQuickPickOpenFromSelection method return false`, async () => {
+      const [loadItemsStub, showStub, setTextStub] = setups.search11();
+      await controller.search();
+
+      assert.equal(loadItemsStub.calledOnce, true);
+      assert.equal(showStub.calledOnce, true);
+      assert.equal(setTextStub.calledOnce, false);
+    });
```

Comment:
```suggestion

    it(`10: should set selected text as quickPick text if shouldSearchSelection
      returns true`, async () => {
      const [setTextStub] = setups.search10();
      await controller.search();

      assert.equal(setTextStub.calledWith("test text"), true);
    });
    
    it(`11: should not invoke quickPick.setText method if shouldSearchSelection
      returns false`, async () => {
      const [setTextStub] = setups.search11();
      await controller.search();

      assert.equal(setTextStub.calledOnce, false);
    });
```

I would suggest to test only whether based on shouldSearchSelection method and shouldSearchSelection method test separately.